### PR TITLE
Use of Tree line tracker in incremental mode

### LINKED ISSUE #426: Improve performance of TextDocument update (in async) with TreeLineTracker
Today we use ListLineTracker (coming from Eclipse Platform) which doesn't support replace method. It means that each time text is changed, the ListLineTracker  recompute the all lines. Even if it's fast (around 40 ms for nasa.xml), when there are several 'textDocument/didChange' with big file like nasa.xml, the completion, foldings, etc is slow.

Ex : you type 10 characters, you wait for 40 * 10 = 400 ms.

To improve this time we can use the TreeLineTracker which supports replace. After doing some test in local, the replace method takes 0 ms, so we win 40 ms for updating TextDocument in asynch mode.

### PR DESCRIPTION
**Pull request by angelozerr**
Fix #426

Signed-off-by: azerr <azerr@redhat.com>

### CODE REVIEW

**Comment by angelozerr** [comment]
To see the performance in action you can execute the [TextDocumentUpdatePerformance ](https://github.com/angelozerr/lsp4xml/pull/448/files#diff-2661998bb406ec92ce6866b5ea0865e2R30). this test inserts a space in the nasa.xml in a loop.

 * with tree line tracker it takes around `40 - 50 ms` (with this PR)
 * without tree line tracker (with listlinetracker which is created each time) it takes around `140 - 300 ms` (you can start TextDocumentUpdatePerformance  in the master by copying/pasting this class in the master).

File: org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/commons/TextDocument.java [hunk 1]
```diff
@@ -155,17 +164,20 @@ public void update(List<TextDocumentContentChangeEvent> changes) {
...
+						lineTracker.replace(startOffset, length, text);
 					}
 					// Update the new text content from the updated buffer
 					setText(buffer.toString());
 				}
+				LOGGER.log(Level.INFO,
+						"Text document content updated in " + (System.currentTimeMillis() - start) + "ms");
 			} catch (BadLocationException e) {
-				// Should never occur.
+				// Should never occurs.
```
[the remark this answers is not in the document]
Comment (angelozerr) [reply org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/commons/TextDocument.java]: @fbricon updated

**Comment by angelozerr** [comment]
> Before auto closing tags wouldn't work, but its really fast now.

Thanks @NikolasKomonen for your feedback!


@fbricon anything else? It should be very cool if this PR could be merged this night.

**Comment by fbricon** [comment]
I've pushed a small change to adjust the logging level with -Dlog.level in the server VM args, to minimize polluting the logs. 
Now I'm ok to merge

**Comment by angelozerr** [comment]
Thanks guys for your review

**Pull request merged**